presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedBucketNodeMap.java

```diff
@@ -15,40 +15,50 @@
 
 import com.facebook.presto.metadata.InternalNode;
 import com.facebook.presto.metadata.Split;
-import com.google.common.collect.ImmutableList;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.function.ToIntFunction;
 
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
 // the bucket to node mapping is fixed and pre-assigned
 public class FixedBucketNodeMap
         extends BucketNodeMap
 {
-    private final List<InternalNode> bucketToNode;
+    private final List<InternalNodeInfo> bucketToNodeInfo;
 
-    public FixedBucketNodeMap(ToIntFunction<Split> splitToBucket, List<InternalNode> bucketToNode)
+    public FixedBucketNodeMap(ToIntFunction<Split> splitToBucket, List<InternalNode> bucketToNode, boolean cacheable)
```

Comment:
I somehow feel this `cacheable` is not needed because everything in fixed map should be cacheable?

Reply:
for a FixedBucketNodeMap, there are situations of `HARD_AFFINITY` and `NO_PREFERENCE`, they both use `FixedBucketNodeMap`, but cacheable for `NO_PREFERENCE` is false, cacheable for `HARD_AFFINITY` is true

Reply:
Aha! Thanks for explaining. So I guess in `FixedBucketNodeMap` it might be reasonable to make `cacheable` a bucket node map level property then, since there won't be update in the mapping.

Reply:
But as you suggested, storing the `cacheable` info in `InternalNodeInfo` would be a more general way, since `DynamicBucketNodeMap` could have a mixed List of `InternalNodeInfo`, we still need to iterate through the list